src/local-development/helpers/origin-checksum.ts

```diff
@@ -0,0 +1,227 @@
+): string[] | null {
+	// Find the checksum record for the component
+	const checksum = findOriginComponent(checksums, componentType, remoteComponentName);
+	if (!checksum) {
+		log('debug', `Not found origin component ${componentType}.`);
+		return null;
+	}
+
+	const outChecksums: string[] = [];
+	// Add the checksum for the specified code type if it exists
+	if (checksum[codeType]) {
+		outChecksums.push(checksum[codeType] || '');
+	}
+
+	// Add the checksum for the JSONC variant of the code type if it exists
+	if (checksum[`${codeType}_jsonc`]) {
+		outChecksums.push(checksum[`${codeType}_jsonc`] || '');
+	}
+
+	// Get the code definition for the component and code type
+	const codeDef = getCodeDef(componentType, codeType);
+	if (!codeDef) {
+		return outChecksums;
+	}
+
+	// Add the checksum for the API code type JSONC variant if it exists
+	if (checksum[`${codeDef.apiCodeType}_jsonc`]) {
+		outChecksums.push(checksum[`${codeDef.apiCodeType}_jsonc`] || '');
+	}
+
+	// Add the checksum for the API code type if it exists
+	if (checksum[codeDef.apiCodeType]) {
+		outChecksums.push(checksum[codeDef.apiCodeType] || '');
+	}
+
+	if (outChecksums.length === 0) {
+		log('warn', `Some field was not found in checksum list componentType=${componentType} codeType=${codeType}, apiCodeType=${codeDef.apiCodeType}`);
+	}
+
+	return outChecksums;
+}
+
+/**
+ * Compares the local data with the origin checksums to determine if they match.
+ *
+ * @param originChecksums - The checksum data from the origin.
+ * @param componentType - The type of the component.
+ * @param componentRemoteName - The name of the remote component.
+ * @param data - The local data to compare.
+ * @returns True if the data matches the origin checksum, false otherwise.
+ */
+export function compareChecksumDeep(
+	originChecksums: Checksum,
+	componentType: AppComponentType,
+	componentRemoteName: string,
+	data: Record<string, any>,
+): boolean {
+	// Find the checksum record for the component
+	const checksum = findOriginComponent(originChecksums, componentType, componentRemoteName);
+	if (!checksum) {
+		return false;
+	}
+
+	// Mapping of local data keys to checksum keys
+	const map: Record<string, string> = {
+		'label': 'label',
+		'description': 'description',
+		'typeId': 'type_id',
+		'actionCrud': 'crud',
+		'crud': 'crud',
+		'connection': 'account_name',
+		'altConnection': 'alt_account_name',
+	};
+
+	// Iterate over the keys in the local data
+	for (const key of Object.keys(data)) {
+		const originKey = map[key];
+		if (!originKey || typeof checksum[originKey] === 'undefined') {
+			log('warn', `Not found mapping for '${componentType}.${key}'.`);
+			return false;
+		}
+
+		// If both values are null, they match
+		if (checksum[originKey] === null && data[key] === null) {
+			continue;
+		}
+
+		// Compare the MD5 hash of the local data value with the checksum
+		if (data[originKey] !== null && md5(data[key].toString()) === checksum[originKey]) {
+			continue;
```

Comment:
Are you sure that `toString()` will exists in all types what can occure as value?
Consider refactor to `String(data[key])`, it is more safe.

Reply:
Checked, OK. :)